fix: Improve clarity around customizations' supported languages (#327)
DeepLcom/api-docs: api-reference/document.mdx

Please be aware that the uploaded document is automatically removed from the server once the translated document has been downloaded. You have to upload the document again in order to restart the translation.

> You may specify the glossary to use for the document translation using the `glossary_id` parameter. **Important:** This requires the `source_lang` parameter to be set and the language pair of the glossary has to match the language pair of the request. If the target language is a variant (e.g., `EN-GB`), a glossary with the base language (e.g., `EN`) as its target language will still be applied.

Reviewer: ```suggestion
You may specify the glossary to use for the document translation using the `glossary_id` parameter. **Important:** This requires the `source_lang` parameter to be set and the language pair of the glossary has to match the language pair of the request. Glossaries created for a base language (e.g. `EN`) can be applied to variants of that language (e.g. `EN-GB`).
```

Reviewer: If you like this wording (or similar), you can apply it below as well

Author: I updated the text here and everywhere else too. I also used the same verbiage for style rule lists too
